exp-player/addon/components/exp-video-physics.js

```diff
@@ -0,0 +1,172 @@
+                    description: 'List of objects specifying video src and type for alternate test videos',
+                    default: []
+                },
+                introSources: {
+                    type: 'string',
+                    description: 'List of objects specifying intro video src and type',
+                    default: []
+                },
+                attnSources: {
+                    type: 'string',
+                    description: 'List of objects specifying attention-grabber video src and type',
+                    default: []
+                },
+                testLength: {
+                    type: 'number',
+                    description: 'Length of test videos in seconds',
+                    default: 20
+                }
+            }
+        },
+        data: {
+            // This video does not explicitly capture any parameters from the userdata:
+            type: 'object',
+            properties: {  // We don't *need* to tell the server about this, but it might be nice to track usage of the setup page
+                doingIntro: {
+                    type: 'boolean',
+                    default: true
+                },
+                doingAttn: {
+                    type: 'boolean',
+                    default: false
+                },
+                useAlternate: {
+                    type: 'boolean',
+                    default: false
+                },
+                timeoutID: {
+                    type: 'number',
+                    default: 0
+                }
+            },
+            required: ['doingIntro', 'doingAttn', 'useAlternate']
+        }
+    },
+    actions: {
+        playNext: function() {
+            window.clearTimeout(this.get('timeoutID'));
+            if (this.get('doingIntro')) { // moving to test video
+                this.set('doingIntro', false);
+                $('#player-video')
+                    .attr('loop', true)
+                    .addClass('test')
+                    .removeClass('intro attn');
+            } else {
+                this.get('next')(); // moving to intro video
+                $('#player-video')
+                    .removeClass('attn test')
+                    .addClass('intro')
+                    .attr('loop', false);
+            }
+        },
+        startVideo: function () {
+            if (!this.get('doingIntro') && !this.get('doingAttn')) {
+                var emberObj = this;
+                var t = window.setTimeout(function(emb) {
+                    emb.send('playNext');
+                    }, emberObj.get('testLength')*1000, emberObj);
+                this.set('timeoutID', t);
+            }
+        },
+        pause: function () {
+            var movie = $('#player-video')[0];
+            var emberObj = this;
+            window.clearTimeout(this.get('timeoutID'));
+            if (!emberObj.get('doingAttn') || !this.checkFullscreen()) { // pausing one of the videos
```

Comment:
Are `doingAttn` and `doingIntro` mutually exclusive conditions? I read this logic like

> If not doing an attention grabber or not fullscreen, show a fullscreen attention grabber. Otherwise if not doing an intro video check if an alternate video is available. If so, show it, otherwise move move. Regardless hide the attention grabber.

I think we may be seeing competing 'pause' events from `onFullScreen` and the event handler for spacebar events. I started tweaking the code but am realizing I don't know exactly how this needs to work.